tests/Microsoft.Identity.Test.LabInfrastructure/LabAuthenticationHelper.cs

```diff
@@ -20,7 +20,7 @@ public static class LabAuthenticationHelper
         
         public static async Task<AccessToken> GetAccessTokenForLabAPIAsync(string labAccessClientId)
         {
-            string[] scopes = new string[] { "https://msidlab.com/.default" };
+            string[] scopes = new string[] { "https://request.msidlab.com/.default" };
```

Comment:
Was this unavoidable?

Reply:
To align with our MISE KPI regarding the security practices around token requests, we removed mutual API access between the RequestMSIDLAB application and IdentityLabsOneAccess application that was using [https://msidlab.com](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmsidlab.com%2F&data=05%7C02%7CGladwin.Johnson%40microsoft.com%7Cb3b6eb3c63a6495c0fdf08dc8a1abbcd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638537094651941926%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=VTXDsnw%2Fc516S8nDhPjqUcxLAL6VY6nyn2T%2BmCQRCaY%3D&reserved=0) as its App ID URL. This was intended to prevent these applications from requesting tokens on behalf of each other.

Post-implementation of these changes, we are no longer able to request tokens for the RequestMSIDLAB using [https://msidlab.com/.default](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmsidlab.com%2F.default&data=05%7C02%7CGladwin.Johnson%40microsoft.com%7Cb3b6eb3c63a6495c0fdf08dc8a1abbcd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638537094651953801%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=Vdl1U%2BeI7WBp9VOeEo%2F5ezmAw0MbmzBJ%2Fb11xIKixzg%3D&reserved=0) i.e. using the second app's scope. 

Given the constraints and the necessity to maintain legacy Lab app compatibility, we cannot reconfigure the RequestMSIDLAB application to use [https://msidlab.com](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmsidlab.com%2F&data=05%7C02%7CGladwin.Johnson%40microsoft.com%7Cb3b6eb3c63a6495c0fdf08dc8a1abbcd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638537094651962471%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=bezGhjZEEVW4XbQ3N%2FalIibwEhEUim2luhSvaMYnPuU%3D&reserved=0) as the App ID URL. 

Resolution: After further testing and evaluations, we have found that utilizing [https://request.msidlab.com/.default](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Frequest.msidlab.com%2F.default&data=05%7C02%7CGladwin.Johnson%40microsoft.com%7Cb3b6eb3c63a6495c0fdf08dc8a1abbcd%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638537094651970109%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=IUeial8MaaMHhtECfXWJkppozH5tUn2Cti2pnTGfkyc%3D&reserved=0) as the scope for token requests resolves the issues without affecting the operation of our legacy lab applications. This scope aligns with our current security and operational configurations and will be our path forward.